crates/tauri/src/plugin.rs

```diff
@@ -104,6 +104,15 @@ pub trait Plugin<R: Runtime>: Send {
   #[allow(unused_variables)]
   fn on_event(&mut self, app: &AppHandle<R>, event: &RunEvent) {}
 
+  /// Callback invoked when the application is performing cleanup before exit.
+  ///
+  /// Plugins can use this hook to perform any process shutdown/cleanup they need
+  /// to do before the runtime exits (for example, killing sidecars or stopping
+  /// background tasks). This is guaranteed to run from the thread performing
+  /// the app cleanup/exit sequence.
+  #[allow(unused_variables)]
+  fn cleanup_before_exit(&mut self, app: &AppHandle<R>) {}
```

Comment:
We would also want to implement this for the `TauriPlugin` so that `tauri::plugin::Builder` could have access to this API